.internal/baseSum.js

```diff
@@ -10,8 +12,11 @@ function baseSum(array, iteratee) {
   let result
 
   for (const value of array) {
-    const current = iteratee(value)
+    let current = iteratee(value)
     if (current !== undefined) {
+      if (typeof current === 'string') {
+        current = baseToNumber(current)
```

Comment:
Just wondering, why don't we call this `stringToNumber` or `parseNumber` Feels more intuitive than `baseToNumber`

Reply:
I think that the `baseToNumber` is more generic, we can convert other types: binary, hexadecimal...